src/interactivity/win32/screenInfoUiaProvider.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "precomp.h"
+
+#include "screenInfoUiaProvider.hpp"
+#include "..\types\WindowUiaProviderBase.hpp"
+
+using namespace Microsoft::Console::Types;
+using namespace Microsoft::Console::Render;
+using namespace Microsoft::Console::Interactivity;
+using namespace Microsoft::Console::Interactivity::Win32;
+
+ScreenInfoUiaProvider::ScreenInfoUiaProvider(_In_ IRenderData* pData,
+                                             _In_ WindowUiaProviderBase* const pUiaParent) :
+    _pUiaParent(THROW_HR_IF_NULL(E_INVALIDARG, pUiaParent)),
+    ScreenInfoUiaProviderBase(THROW_HR_IF_NULL(E_INVALIDARG, pData))
+{
+}
+
+IFACEMETHODIMP ScreenInfoUiaProvider::Navigate(_In_ NavigateDirection direction,
+                                               _COM_Outptr_result_maybenull_ IRawElementProviderFragment** ppProvider)
+{
+    // TODO GitHub 2120: _pUiaParent should not be allowed to be null
+    RETURN_HR_IF(E_NOTIMPL, _pUiaParent == nullptr);
+

```

Comment:
For all of these, you should `RETURN_HR_IF(E_INVALIDARG, ppProvider == nullptr)` to say that the ppProvider arg is invalid if it was null.

You should also ensure that they all set the value of that location to nullptr as their first step before doing the other work. (This one does, some of the ones below do not.)